zcash_client_sqlite/src/lib.rs

```diff
@@ -224,6 +225,39 @@ impl<P: consensus::Parameters + Clone> WalletDb<Connection, P> {
     }
 }
 
+impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletDb<C, P> {
+    pub fn read_transactionally<F, A, E: From<rusqlite::Error>>(&self, f: F) -> Result<A, E>
+    where
+        F: FnOnce(&WalletDb<SqlTransaction<'_>, P>) -> Result<A, E>,
+    {
+        let mut rng = thread_rng();
+        for backoff_ms in [0u64, 1, 2, 4, 8, 16, 32].into_iter().chain(repeat(64)) {
```

Comment:
If this somehow gets into a state where it loops forever, it will sleep an average of 32 ms per iteration, which hopefully should not peg the CPU.